src/client/mapblock_mesh.cpp

```diff
@@ -1195,10 +1177,23 @@ MapBlockMesh::MapBlockMesh(MeshMakeData *data, v3s16 camera_offset):
 	m_enable_shaders = data->m_use_shaders;
 	m_enable_vbo = g_settings->getBool("enable_vbo");
 
-	if (data->m_client->getMinimap()) {
-		m_minimap_mapblock = new MinimapMapblock;
-		m_minimap_mapblock->getMinimapNodes(
-			&data->m_vmanip, data->m_blockpos * MAP_BLOCKSIZE);
+	v3s16 bp = data->m_blockpos;
+	// Only generate minimap mapblocks at even coordinates.
+	if (((bp.X | bp.Y | bp.Z) & 1) == 0 && data->m_client->getMinimap()) {
+		m_minimap_mapblocks.resize(8, nullptr);
+		v3s16 ofs;
+
+		// See also client.cpp for the code that reads the array of minimap blocks.
+		for (ofs.Z = 0; ofs.Z <= 1; ofs.Z++)
+		for (ofs.Y = 0; ofs.Y <= 1; ofs.Y++)
+		for (ofs.X = 0; ofs.X <= 1; ofs.X++) {
+			v3s16 p = (bp + ofs) * MAP_BLOCKSIZE;
+			if (data->m_vmanip.getNodeNoEx(p).getContent() != CONTENT_IGNORE) {
```

Comment:
isn't there a general way to check whether the mapblock is present?
checking a single node against ignore doesn't seem ideal

Reply:
`VoxelManipulator::exists(p)` should do the trick

Reply:
Done

Reply:
Breaks the minimap because blocks with ignore nodes get loaded and produce black rectangles. The situation is better with not loading missing blocks into vmanip (and changes in the mesher), but the fix is incomplete without precise tracking of mapblock state.